packages/sdk/src/conversation/middlewares/cardActionMiddleware.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ActivityTypes,
+  CardFactory,
+  InvokeResponse,
+  MessageFactory,
+  Middleware,
+  StatusCodes,
+  TurnContext,
+} from "botbuilder";
+import { AdaptiveCardResponse, TeamsFxAdaptiveCardActionHandler } from "../interface";
+import { InvokeResponseFactory } from "../invokeResponseFactory";
+
+/**
+ * @internal
+ */
+export class CardActionMiddleware implements Middleware {
+  public readonly actionHandlers: TeamsFxAdaptiveCardActionHandler[] = [];
+  private readonly defaultMessage: string = "Your response was sent to the app";
+
+  constructor(handlers?: TeamsFxAdaptiveCardActionHandler[]) {
+    if (handlers && handlers.length > 0) {
+      this.actionHandlers.push(...handlers);
+    }
+  }
+
+  async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
+    if (context.activity.name === "adaptiveCard/action") {
+      const action = context.activity.value.action;
+      const actionVerb = action.verb;
```

Comment:
Is there any scenario that two cards define the same verb? E.g., a PR review card defines `approve` verb and a release pipeline card also defines `approve` verb?
Or, it's recommended to define unique verbs such as `approve-pr` and `approve-release`?

Reply:
The latter will be recommended in this case :)